### PR TITLE
[FIX] website_sale: restore community ecommerce dashboard

### PR DESCRIPTION
**Pull request by bso-odoo**
Since [1] the community ecommerce dashboard was not rendered anymore. This is because a `t-name` was combined with a `t-extend`, but `t-extend` is ignored by `assetsbundle.py` if a `t-name` is defined.

This commit separates the two aspects:
- a new template with a `t-name` is defined with the actual content of the dashboard, that can be patched by enterprise
- the `t-extend` template just `t-call`s the new template

[1]: https://github.com/odoo/odoo/commit/dcf79f01b33a4f3ddb7e77480f0a7a8be1cd3794

task-2993773

### CODE REVIEW

**Comment by bso-odoo** [comment]
@rdeodoo This PR restores the community ecommerce dashboard, it correctly disappears in enterprise.

**Review by rdeodoo — CHANGES_REQUESTED** [review]
Didn't we say that `t-extend` was deprecated and should be replaced by `t-inherit`?
What's wrong with
```xml
<t t-name="website_sale.dashboard_content" t-inherit="website.dashboard_content"  t-inherit-mode="extension">
        <xpath expr="//div[hasclass('o_website_dashboard_content')]/*[1]" position="before">
```
Instead of
```xml
<t t-name="website_sale.dashboard_content" t-extend="website.dashboard_content">
        <t t-jquery="div.o_website_dashboard_content" t-operation="prepend">
```

Edit: I'd rather use the correct instruction or make `t-extend` fix by the team in charge of it (which they probably won't as deprecated)

**Comment by rdeodoo** [comment]
@bso-odoo I actually found the real issue, see in https://github.com/odoo/odoo/pull/102286 and https://github.com/odoo/enterprise/pull/32464 I added a new commit on top of each PR to solve it.
tl;dr: with `t-inherit` you have to inherit the base template, not the one you want to inherit..
```xml
<!-- module A -->
<template t-name="A.template"/>
<!-- module B -->
<template t-name="B.template" t-inherit="A.template" t-inherit-mode="extension"/>
<!-- module C -->
<template t-name="C.template" t-inherit="B.template" t-inherit-mode="primary"/> <!-- This is primary inherit mode, it needs to inherit B -->
<template t-name="C.template" t-inherit="A.template" t-inherit-mode="extension"/> <!-- This is extension inherit mode, it needs to inherit A, wtf ?! -->
```
